src/workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,134 @@
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# p5 Visualizer Template
+
+[image should go here]
+
+This template can be used as a starting point for designing a p5 visualizer.
+
+## Parameters
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    name = 'p5 Visualizer Template' // the name shown on the visualizer menu
+
+    // external parameters, which the user can choose while creating the
+    // visualizer bundle
+    baseIndex = 0 // the default value of this parameter is zero
+    params = {
+        /** md
+- baseIndex: Where in the sequence to start browsing.
+         **/
+        baseIndex: {
+            value: this.baseIndex,
+            displayName: 'Base index',
+            required: true,
```

Comment:
if you want to force baseIndex to be an integer, you need a `forceType: 'integer',` in here, as TypeScript can't tell the difference between a float and an integer, since it just has number type.

Reply:
Added.